### ISSUE TITLE
Error when creating image attachments

### ISSUE THREAD

**Issue by apanagio** [nautobot/nautobot#4907]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.5
* Python version: 3.11
* Database platform, version: Postgresql 16
* Middleware(s):

### Steps to Reproduce
1. Install Nautobot
2. Create Location
3. Create Imageattachment at location

### Expected Behavior
After image creation to be redirected to the location page

### Observed Behavior
An error appears
```
<class 'AttributeError'>  
Cannot find a URL for XXX (extras.imageattachment)
```

**Comment by gsnider2195** [comment]
I'm seeing this same behavior on demo.nautobot.com. Here's the traceback

```
INFO:nautobot.core.views.generic.ObjectEditView:Created image attachment test (PK: 60f33665-bf32-4adb-b2f1-3d2f2a770fc0)
14:42:32.765 ERROR   django.request :
  Internal Server Error: /dcim/locations/1d0a9604-a511-4684-9b4b-937958f3acf7/images/add/
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 371, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/mixins.py", line 167, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 424, in post
    self.successful_post(request, obj, object_created, logger)
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 399, in successful_post
    msg = format_html('{} <a href="{}">{}</a>', msg, obj.get_absolute_url(), obj)
                                                     ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/models/__init__.py", line 78, in get_absolute_url
    raise AttributeError(f"Cannot find a URL for {self} ({self._meta.app_label}.{self._meta.model_name})")
AttributeError: Cannot find a URL for test (extras.imageattachment)
ERROR:django.request:Internal Server Error: /dcim/locations/1d0a9604-a511-4684-9b4b-937958f3acf7/images/add/
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 371, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/mixins.py", line 167, in dispatch
    return super().dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 424, in post
    self.successful_post(request, obj, object_created, logger)
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/views/generic.py", line 399, in successful_post
    msg = format_html('{} <a href="{}">{}</a>', msg, obj.get_absolute_url(), obj)
                                                     ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/nautobot/core/models/__init__.py", line 78, in get_absolute_url
    raise AttributeError(f"Cannot find a URL for {self} ({self._meta.app_label}.{self._meta.model_name})")
AttributeError: Cannot find a URL for test (extras.imageattachment)
[pid: 135|app: 0|req: 48220/81797] 172.18.0.1 () {66 vars in 1584 bytes} [Mon Dec  4 14:42:32 2023] POST /dcim/locations/1d0a9604-a511-4684-9b4b-937958f3acf7/images/add/ => generated 315877 bytes in 216 msecs (HTTP/1.1 500) 7 headers in 378 bytes (11 switches on core 0)
```

**Comment by glennmatthews** [comment]
Looks similar to the issue fixed by #4868?

**Comment by lampwins** [comment]
Confirmed working in 2.0.6
